docs/api-documentation.md

```diff
@@ -0,0 +1,370 @@
+# API Documentation
+
+This document provides a brief overview of the different API endpoints, their usage and purpose.
+
+## Authentication
+
+This section lists the authentication related requests such as login.
+
+### Login
+
+Allows the user to login into the portal and receive the auth token from the API. Requires the user email and password values.
+
+**User Login Flow**
+
+Here is a step-by-step breakdown of the user login flow:
+
+1. **Client-Side Request:**
+
+   - The client sends a login request to the server.
+
+2. **AuthenticationController Handling:**
+
+   - The `AuthenticationController` captures the login request through the `POST /login` endpoint.
+
+3. **Service Delegation:**
+
+   - The `AuthenticationController` delegates the login process to the `AuthenticationService`.
+
+4. **AuthenticationService Processing:**
+
+   - The `AuthenticationService` is responsible for processing the login request.
+   - It interacts with the `UserRepository` to verify user credentials.
+
+5. **UserRepository Database Interaction:**
+
+   - The `UserRepository` interacts with the database (Entity Framework) to:
+     - Utilize `PasswordSignInAsync` function for password validation provided by `SignInManager` component of Asp.Net Identity.
+     - Retrieve user details using `UserManager` function's provided by Asp.Net Identity.
+
+6. **JWT Token Generation:**
+
+   - Upon successful login, the `AuthenticationService` generates a JWT token using user information.
+
+7. **Response to Client:**
+
+   - The `AuthenticationController` responds to the client with the generated JWT token.
+
+8. **ASP.NET Identity Components Involved:**
+   - `IdentityApplicationUser`: Represents the user entity in the Identity system.
+   - `SignInManager<IdentityApplicationUser>`: Manages sign-in functionality.
+   - `UserManager<IdentityApplicationUser>`: Manages user-related operations.
+   - `JwtSecurityToken`: Represents a JWT token.
+
+The user login flow involves the client sending a login request, which is handled by the `AuthenticationController`. The controller delegates the login process to the `AuthenticationService`, and the user credentials are verified by interacting with the `UserRepository`. If successful, a JWT token is generated and sent back to the client. This flow ensures a secure and structured user login process, utilizing key components provided by ASP.NET Identity and Entity Framework.
+
+**Database Table**
+
+Asp.Net User Table
+![Asp.Net User Table](./images/db_table_aspnetusers.png)
+
+User Table
+
+![User Table](./images/db_table_users.png)
+
+**Endpoint:** `http://localhost:5000/api/auth/register`
+
+**Method: `POST`**
+
+**Body:**
```

Comment:
This part of the document is not structured properly. You should structure your document in a way that it can be listed in the Table of Contents.

Reply:
fixed